src/gui/optionsdialog.cpp

```diff
@@ -230,9 +232,30 @@ OptionsDialog::OptionsDialog(QWidget *parent)
 #if (defined(Q_OS_UNIX) && !defined(Q_OS_MAC)) && !defined(QT_DBUS_LIB)
     m_ui->checkPreventFromSuspend->setDisabled(true);
 #endif
-#if defined(Q_OS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_OS_MAC) || defined(Q_OS_WIN)
     connect(m_ui->checkAssociateTorrents, &QAbstractButton::toggled, this, &ThisType::enableApplyButton);
     connect(m_ui->checkAssociateMagnetLinks, &QAbstractButton::toggled, this, &ThisType::enableApplyButton);
+#endif
+#ifdef Q_OS_WIN
+    if (QSysInfo::WindowsVersion < QSysInfo::WV_VISTA) {
+        m_ui->defaultProgramPanel->hide();
+    }
+    else {
+        m_ui->assocPanel->hide();
+#ifdef WIN_USE_DEFAULT_PROGRAMS
+        connect(m_ui->btnDefProgs, &QPushButton::clicked, this, [this]()
```

Comment:
The first `this` is unnecessary, I believe, And I even wonder how did it pass compilation.

Reply:
>And I even wonder how did it pass compilation.

Why should it fails?
>The first this is unnecessary, I believe,

???

Reply:
Because you wrote `connect(sender, signal, receiver, lambda)`, and moc has to generate a call similar to `(receiver->*lambda)())`, i.e. `(this->*lambda)` which is nonsense.

Reply:
>Because you wrote `connect(sender, signal, receiver, lambda)`

I wrote `connect(sender, signal, context, lambda)`:
```c++
static QMetaObject::Connection QObject::connect(const QObject *sender, PointerToMemberFunction signal, const QObject *context, Functor functor, Qt::ConnectionType type = Qt::AutoConnection);
```

Reply:
Oh, my bad. Did not read this part of the Qt documentation since Qt 5.0.